test_rmw_implementation/test/test_publisher.cpp

```diff
@@ -375,3 +375,60 @@ TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), count_mismatched_subscri
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, subscription_count);
 }
+
+class CLASSNAME (TestPublisherUseLoan, RMW_IMPLEMENTATION)
+  : public CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION)
+{
+protected:
+  using Base = CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION);
+
+  void SetUp() override
+  {
+    Base::SetUp();
+    // Test if Loaning is supported
+    void * msg_pointer{nullptr};
+    rmw_ret_t ret = rmw_borrow_loaned_message(pub, ts, &msg_pointer);
+    if (ret == RMW_RET_UNSUPPORTED) {
+      rmw_reset_error();
+      GTEST_SKIP();
```

Comment:
@Lobotuerk IIRC `GTEST_SKIP()` ends the test, so the `else` clause is redundant (i.e. not necessary).